src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -499,7 +499,7 @@ internal static class CachedReflectionInfo
             typeof(VariableOps).GetMethod(nameof(VariableOps.SetVariableValue), staticFlags);
 
         internal static readonly MethodInfo Utils_IsComObject =
-            typeof(Utils).GetMethod(nameof(Utils.IsComObject), staticFlags, binder: null, types: new Type[] {typeof(object)}, modifiers: null);
+            typeof(Utils).GetMethod(nameof(Utils.IsComObject), staticFlags);
```

Comment:
Seems it is single place where we use the Utils.IsComObject method. Could you please clarify why we can not use Marshal.IsComObject here? Null check?

Reply:
`Marshal.IsComObject` cannot accept null input -- it throws `ArgumentNullException` if the argument is null. `Utils.IsComObject` handles `null` input properly. For the rest call sites where I replaced with `Marshal.IsComObject`, the input object is guaranteed to be non-null.

Reply:
Thanks!